### PR TITLE
TT-12681 DX-1576  added mdcb 2.7 release notes

### PR DESCRIPTION
**Pull request by sredxny**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

Preview Link: https://deploy-preview-5133--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7/
---

### Preview Link

<br>

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Added comprehensive release notes for Tyk MDCB v2.7, including new features, endpoints, and upgrade instructions.
- Introduced new health check endpoints: `/liveness` and `/readiness` for improved monitoring.
- Implemented a new `/config` endpoint for secure, real-time access to MDCB instance configuration.
- Updated API definition for compatibility with Tyk Gateway 5.5.0.
- Listed third-party dependencies and their tested and compatible versions.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>version-2.7.md</strong><dd><code>Added release notes for Tyk MDCB v2.7</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md

<li>Added release notes for Tyk MDCB v2.7.<br> <li> Included details on new endpoints: <code>/liveness</code>, <code>/readiness</code>, and <code>/config</code>.<br> <li> Provided upgrade instructions and release highlights.<br> <li> Listed third-party dependencies and their versions.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5133/files#diff-ee9bf0e4c47e6f6f8c0390eace804ba56321c2de5e2acc820da7c4a00cffdda1">+109/-0</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.
+
+#### Deprecations
+There are no deprecations in this release, however with the introduction of new healthcheck endpoints we encourage customers to start using the new `/liveness` and `/readiness` endpoints and avoid using the old `/health` endpoint.
+
+###### Changes in MDCB v2.7.0:
+- Added `/liveness` endpoint for quick checks on MDCB application status.
+- Implemented `/readiness` endpoint to detail status of critical components and dependencies.
+- Introduced `/config` endpoint for secure, real-time access to MDCB instance configuration.
```
Comment (caroltyk) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:38]:
```suggestion
- Introduced `/config` and `/env` endpoint for secure, real-time access to MDCB instance configuration.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+1. `/liveness`: This endpoint provides a quick check to determine if the MDCB application is alive and running.
+2. `/readiness`: This endpoint offers a detailed status of components and dependencies required for MDCB to serve traffic. It includes status checks for:
+    - Database connectivity
+    - Redis connectivity
+    - RPC server status
+
+These new endpoints allow for more granular monitoring of MDCB's operational status, enabling quicker identification and resolution of potential issues.
+
+##### New Configuration Access Endpoint
+A new `/config` endpoint has been implemented, allowing developers to access the current configuration state of the MDCB instance in real-time. This feature provides:
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]:
```suggestion
Two new `/config` and `/env` endpoints has been implemented, allowing developers to access the current configuration state of the MDCB instance in real-time. This feature provides:
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+###### Changes in MDCB v2.7.0:
+- Added `/liveness` endpoint for quick checks on MDCB application status.
+- Implemented `/readiness` endpoint to detail status of critical components and dependencies.
+- Introduced `/config` endpoint for secure, real-time access to MDCB instance configuration.
+- Updated API definition for compatibility with Tyk Gateway 5.5.0.
+
+###### Recommendations for users:
+
+- Migrate to new healthcheck endpoints in order to get more detailed information.
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]:
```suggestion
- Migrate to new health check endpoints in order to get more detailed information. For Kubernetes users, use Helm Charts v1.6 to upgrade MDCB to set liveness and readiness probes of MDCB deployment to the new health check endpoints.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+#### Upgrade instructions
+If you are using a 2.6.x version, we advise you to upgrade ASAP to this latest release. If you are on an older version, you should skip 2.6.0 and upgrade directly to this release.
+
+#### Release Highlights
+Two new health check endpoints have been added to improve monitoring and diagnostics:
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]: Add a heading
##### New Health check probes

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md (hunk 1 above)
Comment (caroltyk) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:38]: I think we can move this part to #### Changelog {#Changelog-v2.7.0} section

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+#### Deprecations
+There are no deprecations in this release, however with the introduction of new healthcheck endpoints we encourage customers to start using the new `/liveness` and `/readiness` endpoints and avoid using the old `/health` endpoint.
+
+###### Changes in MDCB v2.7.0:
+- Added `/liveness` endpoint for quick checks on MDCB application status.
+- Implemented `/readiness` endpoint to detail status of critical components and dependencies.
+- Introduced `/config` endpoint for secure, real-time access to MDCB instance configuration.
+- Updated API definition for compatibility with Tyk Gateway 5.5.0.
+
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]: I think we can remove this part put it into #### Changelog {#Changelog-v2.7.0} section

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  ``` 
+
+#### Changelog {#Changelog-v2.7.0}
+
+##### Added
+<ul>
+   <li>
+ <details>
+ <summary>Retrieve information of all the connected data plane nodes</summary>
+   Adding a `/dataplanes` endpoint that offers a comprehensive view of all data plane nodes connected to MDCB. This endpoint provides crucial metadata and status information for each connected node, enabling efficient monitoring and troubleshooting. It requires an administrative key provided in the `x-tyk-authorization` header for access, ensuring secure and controlled usage. Successful requests return an array of node details, including node ID, API key, group ID, version, TTL, tags, health status, API statistics, and host details.
```
Comment (carolwkc) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]:
```suggestion
 <summary> Added `/liveness` endpoint for quick checks on MDCB application status </summary>
   Added `/liveness` endpoint that report if MDCB is running. It returns status 200 if MDCB is alive. It returns status 503 if MDCB is not operational. In that case, a restart is recommended. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
    </details>
  </li>
   <li>
  <details>
   <summary> Implemented `/readiness` endpoint to detail status of critical components and dependencies </summary>
   Added `/readiness` endpoint that report if MDCB is ready to serve request. It returns status 200 if MDCB is ready. It returns status 503 if MDCB or one of the dependencies is not ready. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
      </details>
  </li>
   <li>
  <details>
   <summary> Introduced `/config` endpoint for secure, real-time access to MDCB instance configuration </summary>
   Added `/config` endpoint that returns MDCB instance configuration in JSON format. It requires an administrative key provided in the `x-tyk-authorization` header for access, ensuring secure and controlled usage. Successful requests return MDCB JSON configurations with passwords and sensitive information redacted.
      </details>
  </li>
   <li>
  <details>
   <summary> Introduced `/env` endpoint for secure, real-time access to MDCB instance configuration </summary>
      Added `/env` endpoint that returns MDCB instance configuration as a list of environment variable keys and values. It requires an administrative key provided in the `x-tyk-authorization` header for access, ensuring secure and controlled usage. Successful requests returns a list of environment variable keys and values with passwords and sensitive information redacted.
      </details>
  </li>
```

**Comment by carolwkc** [comment]
Thank you @sredxny for raising this. I've added some comments.

**Comment by sredxny** [comment]
@caroltyk thanks for the review. I pushed the changes

**Review by carolwkc — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+Licensed Protected Product
+
+*This page contains all release notes for version 2.7 displayed in reverse chronological order*
+
+## Support Lifetime
+Our minor releases are supported until our next minor comes out.
+
+## 2.7.0 Release Notes
+
+##### Release date 01 August 2024
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]: Remember to add release date

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+   <li>
+ <details>
+ <summary> Added `/liveness` endpoint for quick checks on MDCB application status </summary>
+   Added `/liveness` endpoint that report if MDCB is running. It returns status 200 if MDCB is alive. It returns status 503 if MDCB is not operational. In that case, a restart is recommended. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
+    </details>
+  </li>
+   <li>
+  <details>
+   <summary> Implemented `/readiness` endpoint to detail status of critical components and dependencies </summary>
+   Added `/readiness` endpoint that report if MDCB is ready to serve request. It returns status 200 if MDCB is ready. It returns status 503 if MDCB or one of the dependencies is not ready. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]:
```suggestion
   Added `/readiness` endpoint that reports if MDCB is ready to serve request. It returns status 200 if MDCB is ready. It returns status 503 if MDCB or one of the dependencies is not ready. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+  ``` 
+
+#### Changelog {#Changelog-v2.7.0}
+
+##### Added
+<ul>
+   <li>
+ <details>
+ <summary> Added `/liveness` endpoint for quick checks on MDCB application status </summary>
+   Added `/liveness` endpoint that report if MDCB is running. It returns status 200 if MDCB is alive. It returns status 503 if MDCB is not operational. In that case, a restart is recommended. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]:
```suggestion
   Added `/liveness` endpoint that reports if MDCB is running. It returns status 200 if MDCB is alive. It returns status 503 if MDCB is not operational. In that case, a restart is recommended. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
```

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+   <li>
+  <details>
+   <summary> Implemented `/readiness` endpoint to detail status of critical components and dependencies </summary>
+   Added `/readiness` endpoint that report if MDCB is ready to serve request. It returns status 200 if MDCB is ready. It returns status 503 if MDCB or one of the dependencies is not ready. For more details, see [MDCB Health check]({{<ref "tyk-multi-data-centre/setup-controller-data-centre#health-check">}}) section.
+      </details>
+  </li>
+   <li>
+  <details>
+   <summary> Introduced `/config` endpoint for secure, real-time access to MDCB instance configuration </summary>
+   Added `/config` endpoint that returns MDCB instance configuration in JSON format. It requires an administrative key provided in the `x-tyk-authorization` header for access, ensuring secure and controlled usage. Successful requests return MDCB JSON configurations with passwords and sensitive information redacted.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:89]: Add links to config and env endpoint sections in docs for the config and env endpoint changelog items, similar to what has been done for the healthcheck endpoints

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x | 4.4.x, 5.0.x, 6.0.x, 7.0.x | Used by MDCB | 
+| [PostgreSQL](https://www.postgresql.org/download/)         | 12.x - 16.x LTS        | 12.x - 16.x            | Used by MDCB | 
+
+Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.
+
+#### Deprecations
+There are no deprecations in this release, however with the introduction of new healthcheck endpoints we encourage customers to start using the new `/liveness` and `/readiness` endpoints and avoid using the old `/health` endpoint.
+
+###### Recommendations for users:
+
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:34]: Add links to healthcheck endpoint docs here and in release highlights section

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
+---
+title: Tyk MDCB v2.7 Release Notes
+description: "Tyk Multi Data-Center Bridge v2.7 release notes"
+tags: ["release notes", "MDCB", "Tyk Multi Data-Center", "Tyk Multi Data-Center", "v2.7", "2.7"]
+aliases:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]: Do we need aliases (redirects for a url) for a new content file?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Thanks @sredxny PR LGTM I have made minor suggestions relating to added links and some and some minor grammar suggestions

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md (hunk 12 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md]: removed

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md (hunk 10 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:89]: added

File: tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md (hunk 11 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-enterprise-mdcb/release-notes/version-2.7.md:34]: added

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**